codegen/runner/runner.go

```diff
@@ -92,15 +92,16 @@ func main() {
 		deputyReqHeader = config.MustGetString("deputyReqHeader")
 	}
 	options := &codegen.PackageHelperOptions{
-		RelThriftRootDir:       config.MustGetString("thriftRootDir"),
-		RelTargetGenDir:        config.MustGetString("targetGenDir"),
-		RelMiddlewareConfigDir: config.MustGetString("middlewareConfig"),
-		AnnotationPrefix:       config.MustGetString("annotationPrefix"),
-		GenCodePackage:         config.MustGetString("genCodePackage"),
-		CopyrightHeader:        string(copyright),
-		StagingReqHeader:       stagingReqHeader,
-		DeputyReqHeader:        deputyReqHeader,
-		TraceKey:               config.MustGetString("traceKey"),
+		RelThriftRootDir:         config.MustGetString("thriftRootDir"),
+		RelTargetGenDir:          config.MustGetString("targetGenDir"),
+		RelMiddlewareConfigDir:   config.MustGetString("middlewareConfig"),
+		RelMandatoryMidConfigDir: config.MustGetString("mandatoryMiddlewareConfig"),
```

Comment:
We should make such a list optional, i.e., an application does not necessarily have mandatory middlewares (or middlewares at all).

Reply:
Correct, we actually see problems with that and have to workout with that by having an empty dir. If you can fixed that along the way, that would be awesome!

Reply:
Fixed this. Had to change a test along the way (`template_test.go`) because the test relied on the default value of `middlewareConfig` being "./middlewares".